### PR TITLE
Implement `batch_inverse_and_mul`

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Implement batch inverse and mul, and add tests for both batch inverse and batch inverse & mul. 

closes: #98 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by Pratyush** [comment]
Thanks for the PR! To get the test to compile you need to add a `dev-dependency` on `ark-test-curves`.

**Comment by ValarDragon** [comment]
Thanks! Just realized I was testing it incorrectly locally (forgot to use all features)

**Comment by ValarDragon** [comment]
I'm running into an error that doesn't really make sense to me:

```
484 | pub fn batch_inversion<F: Field>(v: &mut [F]) {
    |                           ----- required by this bound in `fields::batch_inversion`
...
559 |         batch_inversion::<Fr>(&mut random_coeffs_inv);
    |                               ^^^^^^^^^^^^^^^^^^^^^^ the trait `fields::Field` is not implemented for `ark_ff::fields::models::Fp256<FrParameters>`
```

Not sure how to fix it

**Comment by Pratyush** [comment]
> I'm running into an error that doesn't really make sense to me:
> 
> ```
> 484 | pub fn batch_inversion<F: Field>(v: &mut [F]) {
>     |                           ----- required by this bound in `fields::batch_inversion`
> ...
> 559 |         batch_inversion::<Fr>(&mut random_coeffs_inv);
>     |                               ^^^^^^^^^^^^^^^^^^^^^^ the trait `fields::Field` is not implemented for `ark_ff::fields::models::Fp256<FrParameters>`
> ```
> 
> Not sure how to fix it

Hmm this is probably because of some circular dependency issues: `test-curves` is using a different "version" of `ff` in the dependency tree than `ff`'s tests are. Ugh this is annoying. If you'd like you can just copy paste bls12_381's `Fr` into a private module of `ark-ff` (and similarly for `ark-ec`).

**Comment by ValarDragon** [comment]
Oh cool, that worked!

File: ff/src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ pub use self::bytes::*;
 pub mod fields;
 pub use self::fields::*;
 
+// This is only used for testing.
+mod test_field;
```
Comment (Pratyush) [inline ff/src/lib.rs:29]:
```suggestion
#[cfg(test)]
mod test_field;
```

**Comment by Pratyush** [comment]
@ValarDragon Is this ready to merge?

**Comment by ValarDragon** [comment]
Yes!

**Comment by ValarDragon** [comment]
Thanks for the help in debugging this!

**Comment by Pratyush** [comment]
np, thanks for the PR!

**Pull request merged**